elastic/kibana
# [Security Solution][Detections] -Fixes rule edit flow bug with max_signals (#92748)

## Summary

Fixes a bug where max_signals was being reverted to it's default value when the rule was edited via the UI. 

This PR fixes that and adds tests around the various scenarios.

To test, follow the below steps:
1) `cd x-pack/plugins/security_solution/server/lib/detection_engine/scripts`
2) `./post_rule.sh ./rules/queries/query_with_max_signals.json` (creates a rule with max_signals of 500)
3) Via UI, enable/disable rule
  -  `./get_rules.sh` - check to see that rule continues to have `max_signals: 500`
4) Via UI, add an action to prepackaged rule
  -  `./get_rules.sh` - check to see that rule continues to have `max_signals: 500`
5) Via UI, edit rule
  -  `./get_rules.sh` - check to see that rule continues to have `max_signals: 500`
6) Via UI, add an exception
  -  `./get_rules.sh` - check to see that rule continues to have `max_signals: 500`

NOTE: The exceptions flow was not previously breaking `max_signals` but I added it as something to test as it had previously caused some issues on the `PATCH`. 

### Checklist
